packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte

```diff
@@ -95,6 +98,12 @@
     return errors
   }
 
+  $: isManyToMany =
+    fromRelationship?.relationshipType === RelationshipTypes.MANY_TO_MANY
+  $: isManyToOne =
+    fromRelationship?.relationshipType === RelationshipTypes.MANY_TO_ONE
+  $: fromPrimary =
+    !fromPrimary && fromTable ? fromTable.primary[0] : fromPrimary
```

Comment:
I have no idea how svelte handles this lol. Is this not like a recursive assignment? Since the reactive statement references itself? And then it's a reactive assignment but also bound to a form field. I'm guessing it works since the PR is up but the look of it scares me 😆 

Personally I would make `fromPrimary` just be a normal variable, then you can have a reactive snippet to set it to something else when required.

So like:
```js
let fromPrimary
$: {
  if (!fromPrimary && fromTable) {
    fromPrimary = fromTable.primary[0]
  }
}
```

Reply:
Thanks for suggestion, yeah wasn't sure how this would work but it seemed to work fine so left it in - will change to your suggestion.